arduino/libraries/WiFi/src/WiFi.cpp

```diff
@@ -604,7 +611,9 @@ void WiFiClass::init()
   sntp_setservername(1, (char*)"1.pool.ntp.org");
   sntp_setservername(2, (char*)"2.pool.ntp.org");
   sntp_init();
-
+  uint8_t mac[6];
+  esp_wifi_get_mac(ESP_IF_WIFI_STA, mac);
+  sprintf(_defaultHostname, "arduino-%.2x%.2x", mac[4], mac[5]);
```

Comment:
Is there a way to re-use the default hostname generation from above? Also why is it defaulted in `disconnect` don't we need something on first boot too?

Reply:
mmm you mean like an API from the WifiNINA library? we could implement a setDefaultHostname API, like the setHostname, and we can call it in disconnect and from above too, what do you think @sandeepmistry?